crates/chainio/clients/fireblocks/src/lib.rs

```diff
@@ -6,7 +6,231 @@
+
+            match accounts_result {
+                Ok(accounts) => Ok(accounts.vault_accounts()),
+                Err(e) => Err(e),
+            }
+        } else {
+            // Already checking if vault account is available in above if condition, so using expect
+            Ok(self
+                .vault_account
+                .as_ref()
+                .expect("Vault account not found")
+                .to_vec())
+        }
+    }
+
+    /// get whitelisted account for the particular address
+    pub async fn get_whitelisted_account(
+        &mut self,
+        address: Address,
+    ) -> Result<WhitelistedAccount, FireBlockError> {
+        let asset_id_some = ASSET_ID_BY_CHAIN.get(&self.chain_id);
+
+        match asset_id_some {
+            Some(asset_id) => {
+                let whitelisted_account_some = self.whitelisted_accounts.get(&address);
+
+                match whitelisted_account_some {
+                    Some(mut whitelisted_accounts) => {
+                        let accounts_result = self.fireblocks_client.list_external_accounts().await;
+
+                        match accounts_result {
+                            Ok(accounts) => {
+                                for account in accounts.iter() {
+                                    for asset in account.assets.iter() {
+                                        if asset.address.as_ref().unwrap().eq(&address.to_string())
+                                            && asset.status.as_ref().unwrap().as_str() == "APPROVED"
+                                            && *asset.id.as_ref().unwrap() == *asset_id
+                                        {
+                                            self.whitelisted_accounts
+                                                .insert(address, account.clone());
+                                            whitelisted_accounts = account;
+                                            return Ok(whitelisted_accounts.clone());
+                                        }
+                                    }
+                                }
+
+                                Ok(whitelisted_accounts.clone())
+                            }
+                            Err(e) => Err(e),
+                        }
+                    }
+                    None => Err(FireBlockError::AccountNotFoundError(address.to_string())),
+                }
+            }
+            None => Err(FireBlockError::AssetIDError(self.chain_id.to_string())),
+        }
+    }
+
+    /// get whitelisted contract
+    pub async fn get_whitelisted_contract(
+        &mut self,
+        address: Address,
+    ) -> Result<WhitelistedContract, FireBlockError> {
+        let asset_id_some = ASSET_ID_BY_CHAIN.get(&self.chain_id);
+        match asset_id_some {
+            Some(asset_id) => {
+                let contract_some = self.whitelisted_contracts.get(&address);
+
+                match contract_some {
```

Comment:
the _some suffix is weird, because it might be None, please remove here and elsewhere

Reply:
then below you can overload the name with contract again, as you don't need the _some anymore